### PR TITLE
cc7

### PR DESCRIPTION
**Pull request by nojadzworld**
- [ x] Top-level README “Table of Contents” is updated
 - [x ] README for this challenge is complete
       - [ x] Summary, Description, Approach & Efficiency, Solution
       - [ x] Picture of whiteboard
       - [x ] Link to code
 - [x ] Feature tasks for this challenge are completed
 - [ ] Unit tests written and passing
       - [x ] “Happy Path” - Expected outcome
       - [x ] Expected failure
       - [x ] Edge Case (if applicable/obvious)

### CODE REVIEW

File: java/datastructures/lib/src/main/java/datastructures/linkedlist/LinkedList.java [hunk 1]
```diff
@@ -94,6 +94,26 @@ public boolean includes(int value) {
...
 
+  public int kthFromEnd(int k) {
+
+    if (head == null) {
+      throw new IllegalStateException("Linked List is empty");
+    }
+    Node slow = head;
+    Node fast = head;
+    for (int i = 0; i < k; i++) {
+      if (fast == null) {
```
Comment (lucy-gelderloos) [inline java/datastructures/lib/src/main/java/datastructures/linkedlist/LinkedList.java:105]: If `fast` is already null going into this if statement, the if statement will throw a null pointer exception and fail. You'll need to find a way to check whether `fast.next` is null and go from there (same thing for the while statement on line 110).
